File: app/src/main/java/org/amoustakos/boilerplate/injection/module/FragmentModule.java
package org.amoustakos.boilerplate.injection.module;

import android.app.Fragment;
import android.content.Context;

import org.amoustakos.boilerplate.injection.ActivityContext;

import dagger.Module;
import dagger.Provides;


@Module
public class FragmentModule {
    private Fragment mFragment;

    public FragmentModule(Fragment fragment) {
        mFragment = fragment;
    }


    @Provides
    @ActivityContext
    Context providesContext() {
        return mFragment.getActivity();
    }

}

File: app/src/main/java/org/amoustakos/boilerplate/injection/module/injectors/FragmentModule.kt
package org.amoustakos.boilerplate.injection.module.injectors

import android.content.Context
import androidx.fragment.app.Fragment
import dagger.Module
import dagger.Provides
import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext


@Module
class FragmentModule(
		private val mFragment: androidx.fragment.app.Fragment
) {

	@Provides
	@ActivityContext
	internal fun providesContext(): Context? = mFragment.activity

}
